vortex-flatbuffers/flatbuffers/vortex-file/footer.fbs

```diff
@@ -1,34 +1,81 @@
 include "vortex-layout/layout.fbs";
 
-// [file layout]
-/// The `FileLayout` stores the root `Layout` as well as location information for each referenced segment.
-table FileLayout {
-    root_layout: Layout;
+// [postscript]
+/// The `Postscript` is guaranteed by the file format to never exceed
+/// 65528 bytes (i.e., u16::MAX - 8 bytes) in length, and is immediately
+/// followed by an 8-byte `EndOfFile` struct.
+table Postscript {
+    /// The encodings used within the file
+    encodings: [EncodingVTable];
+    /// The layouts used within the file
+    layouts: [LayoutVTable];
+    /// A FlatBuffer of type vortex_dtype::DType
+    dtype: Segment;
+    /// A FlatBuffer of type vortex_layout::Layout
+    layout: Segment;
+    /// A FlatBuffer of type SegmentMap
+    segment_map: Segment;
+    /// A FlatBuffer of type Schemes
+    schemes: Segment;
+}
+// [postscript]
+
+// [segment_map]
+table SegmentMap {
     segments: [Segment];
 }
 
 /// A `Segment` acts as the locator for a buffer within the file.
-struct Segment {
+table Segment {
     offset: uint64;
     length: uint32;
-    alignment_exponent: uint8;
-    // These two fields are reserved for future use and act as pointers
-    // into `FileLayout::compression_schemes` and `FileLayout::encryption_schemes`
-    // respectively. They are not used in the current version of the file format.
-    _compression: uint8;
-    _encryption: uint16;
+    alignment_exponent: uint8 = null;
+    compression_scheme: uint32 = null;
+    encryption_scheme: uint32 = null;
 }
-// [file layout]
+// [segment_map]
 
-// [postscript]
-/// The `Postscript` is guaranteed by the file format to never exceed
-/// 65528 bytes (i.e., u16::MAX - 8 bytes) in length, and is immediately
-/// followed by an 8-byte `EndOfFile` struct.
-table Postscript {
-    dtype: Segment;
-    file_layout: Segment;
+// [schemes]
+
+/// Schemes essentially acts as dictionary encoding for the different encryption and compression
+/// configurations used within the file. Each segment can optionally point into these arrays.
+table Schemes {
+    /// The encryption schemes used within the file
+    encryption_schemes: [EncryptionScheme];
+    /// The compression schemes used within the file
+    compression_schemes: [CompressionScheme];
 }
-// [postscript]
+
+/// Defines an encryption scheme used within the file
+table EncryptionScheme {
+    key_id: string;
+    algorithm: EncryptionAlgorithm;
```

Comment:
maybe should also be a string?

Reply:
expanding - seems like organizations like NIST give algorithms [standard names](https://csrc.nist.gov/projects/computer-security-objects-register/algorithm-registration), and we probably want to stick to that.